packages/liveblocks-react/src/room.tsx

```diff
@@ -1006,13 +1006,7 @@ export function createRoomContext<
     throw innerError;
   }
 
-  const requestsCache: Map<
-    string,
-    {
-      promise: Promise<any> | null;
```

Comment:
We store the `getThreads` and `getRoomNotificationSettings` promise for query keys but do not use it anywhere. I do not believe we plan to use it in future either, so removing this to an easier to understand map of `query key` -> `subscribers count`.

Reply:
I just realized that we're not keeping an instance of the initial promise anymore and we don't return it for suspense. 

My understanding is that it's a requirement to make sure suspense does not regenerate a promise at every render. And I think this is also recommended when using `use`.

Am I correct? If yes, we might have a subtle suspense bug.

Reply:
> My understanding is that it's a requirement to make sure suspense does not regenerate a promise at every render. And I think this is also recommended when using `use`.

Oh, interesting. Yes, it would make sense to require this from React's perspective, i.e. be able to render a component whenever it needs to for whatever reason, without having to worry about creating a ton of promises as a by-product.

> If yes, we might have a subtle suspense bug.

Do you mean in this new code? Or in our `useSuspendUntil*` functions where we're currently throwing those promises? We are indeed creating a [new promise instance every time](https://github.com/liveblocks/liveblocks/blob/c9deed316115ef76ee8c16aeb337e03bb6d45650/packages/liveblocks-react/src/room.tsx#L863-L865) the function would get executed — even though each such instance subscribes to and waits for the same event, so they'll all resolve at the same moment eventually. (Same [here](https://github.com/liveblocks/liveblocks/blob/c9deed316115ef76ee8c16aeb337e03bb6d45650/packages/liveblocks-react/src/room.tsx#L881-L884).)

That said, I just added in a counter to track how often this actually happens, and it looks like this isn't a pressing issue in practice right now. For each such instance, I found at most 2 outstanding promises, so it looks like React will not re-render that component internally again until the promise resolves. At least in my testing. Maybe there are situations where it would actually keep rendering this component, even though it's already suspended.

Either way, maybe a good idea to be ahead of that and change that code to always return the already-existing promise, using the `use()` polyfill you linked in chat.